### PR TITLE
Hidden Path Namespace for Subgraphs

### PR DESCRIPTION
**Pull request by alindmanUnity**
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?
Adding a check to prevent certain subgraphs from populating the Create Node menu. This behaves exactly like "Hidden/" paths in HLSL shaders. 

---
### Release Notes
Please add any useful notes about the feature/fix that might be helpful for others.
Only changes single file, Search Window Provider. 
Currently only filters out "Hidden" in Pascal case. "HIDDEN" or "hidden" will populate the menu. This behavior matches existing behavior for HLSL shaders. 

---
### Testing status
**Katana Tests**: First off we need to make sure the Katana SRP tests are green?
Katan Green

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [x] Checked new UI names with UX convention
- [x] Tested UI multi-edition + Undo/Redo
- [x] C# and shader warnings (supress shader cache to see them)
- Other: 

**Automated Tests**: What did you setup?

Any test projects to go with this to help reviewers?

---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High?
Low - small line change, one file only

**Halo Effect**: None, Low, Medium, High?
Low - users shouldn't already have "Hidden" in paths to suddenly see subgraphs disappear. 

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.shadergraph/Editor/Drawing/SearchWindowProvider.cs [hunk 1]
```diff
@@ -21,6 +21,7 @@ class SearchWindowProvider : ScriptableObject, ISearchWindowProvider
         public bool nodeNeedsRepositioning { get; set; }
         public SlotReference targetSlotReference { get; private set; }
         public Vector2 targetPosition { get; private set; }
+        private static string s_HiddenFolderName => "Hidden";
```
Comment (pbbastian) [inline com.unity.shadergraph/Editor/Drawing/SearchWindowProvider.cs]:
```suggestion
        private const string k_HiddenFolderName => "Hidden";
```

File: com.unity.shadergraph/Editor/Drawing/SearchWindowProvider.cs [hunk 2]
```diff
@@ -21,6 +21,7 @@ class SearchWindowProvider : ScriptableObject, ISearchWindowProvider
         public bool nodeNeedsRepositioning { get; set; }
         public SlotReference targetSlotReference { get; private set; }
         public Vector2 targetPosition { get; private set; }
+        private const string k_HiddenFolderName => "Hidden";
```
Comment (zadi0648) [inline com.unity.shadergraph/Editor/Drawing/SearchWindowProvider.cs]: Should be '='

File: com.unity.shadergraph/Editor/Drawing/SearchWindowProvider.cs [hunk 3]
```diff
@@ -88,9 +89,10 @@ public List<SearchTreeEntry> CreateSearchTree(SearchWindowContext context)
                 {
                     AddEntries(node, new string[1] { asset.name }, nodeEntries);
                 }
-                else
+                
+                var title = node.subGraphData.path.Split('/').ToList();
+                if (title[0] != s_HiddenFolderName)
```
Comment (zadi0648) [inline com.unity.shadergraph/Editor/Drawing/SearchWindowProvider.cs]: 's_HiddenFolderName' does not match 'k_HiddenFolderName'

**Comment by zadi0648** [comment]
At first I thought I found a bug in master but I realized I was on your branch and this only reproduces on your branch: giving a sub graph no group results in it being in the creation tree twice. Once at level 1, a 2nd on at level 2 with the parent being "   ".
AlexHiddenPathBug0

File: com.unity.shadergraph/Editor/Drawing/SearchWindowProvider.cs (hunk 2 above)
Comment (alindmanUnity) [reply com.unity.shadergraph/Editor/Drawing/SearchWindowProvider.cs]: => is a new flag for assigning a get/set, I'm pretty sure it should stay that way

**Pull request merged**